src/api/plugins/tdbp/warren_tdbp/indicators.py

```diff
@@ -0,0 +1,530 @@
+    ):
+        """Initialize Scores indicator."""
+        super().__init__(
+            course_id=course_id,
+            student_id=student_id,
+            until=until,
+            totals=totals,
+            average=average,
+            sliding_window_min=sliding_window_min,
+            active_actions_min=active_actions_min,
+            dynamic_cohort_min=dynamic_cohort_min,
+        )
+
+    def get_lrs_query(self):
+        """Construct the LRS query for statements which object is the course.
+
+        WARNING: this method is used only for key cache computing as the LRS
+        query requires to look for statements which object is related to the course.
+        """
+        return LRSQuery.construct(
+            query={
+                "activity": self.course_id,
+                "until": datetime.combine(self.until, time.min).isoformat(),
+            }
+        )
+
+    async def compute(self) -> ScoresIndicator:
+        """Return cohort scores for active actions."""
+        sliding_window_indicator = SlidingWindow(
+            course_id=self.course_id, until=self.until
+        )
+        sliding_window = await sliding_window_indicator.compute()
+
+        active_actions = pd.DataFrame(
+            [action.dict() for action in sliding_window.active_actions]
+        )
+        active_actions.set_index("iri", inplace=True)
+        course_cohort_indicator = Cohort(course_id=self.course_id, until=self.until)
+        course_cohort = await course_cohort_indicator.compute()
+
+        if self.student_id:
+            if self.student_id not in course_cohort.keys():
+                scores = {
+                    self.student_id: [
+                        -action.activation_rate
+                        for action in sliding_window.active_actions
+                    ]
+                }
+                return ScoresIndicator(
+                    actions=sliding_window.active_actions, scores=scores
+                )
+
+            else:
+                course_cohort = {self.student_id: course_cohort[self.student_id]}
+
+        # Get unique activities
+        unique_activities = sorted(
+            {
+                activity
+                for activities in course_cohort.values()
+                for activity in activities
+            }
+        )
+
+        # Create a DataFrame with 0s and 1s
+        df = pd.DataFrame(
+            {
+                activity: [
+                    1 if activity in course_cohort[student] else 0
+                    for student in course_cohort
+                ]
+                for activity in unique_activities
+            },
+            index=course_cohort.keys(),
+        )
+
+        def compute_students_score(column: pd.Series, action_id, actions: pd.DataFrame):
+            """Apply a transformation on a column."""
+            score = actions.loc[action_id]["activation_rate"]
+            return column.apply(lambda x: float(score) if x == 1 else -float(score))
```

Comment:
@wilbrdt I construct a dataframe based on active actions, I wanted to store 1 and 0 value to avoid saving too big values. Storing actions seem useless for me ! Let's discuss it if necessary

Reply:
My suggestion was a bit picky, and I was pointing that we don't need to write 1s and 0s (`1 if activity in course_cohort[student] else 0`), as we can get a direct boolean value with `activity in course_cohort[student]`.
But either way I'm fine with it!